core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -723,7 +723,7 @@ void populate1() {
           NullPolicy.ARG0);
       defineMethod(CONCAT_WS_MSSQL,
           BuiltInMethod.MULTI_STRING_CONCAT_WITH_SEPARATOR.method,
-          NullPolicy.NONE);
+          NullPolicy.STRICT);
```

Comment:
This one seems wrong too

Reply:
I think this is the same problem. The null guard is for the array references, not the contents of the array.